examples/ex9p.cpp

```diff
@@ -344,9 +344,9 @@ int main(int argc, char *argv[])
    m->AddDomainIntegrator(new MassIntegrator);
    k->AddDomainIntegrator(new ConvectionIntegrator(velocity, -1.0));
    k->AddInteriorFaceIntegrator(
-      new TransposeIntegrator(new DGTraceIntegrator(velocity, 1.0, -0.5)));
+      new NonconservativeDGTraceIntegrator(velocity, -1.0, -0.5));
    k->AddBdrFaceIntegrator(
-      new TransposeIntegrator(new DGTraceIntegrator(velocity, 1.0, -0.5)));
+      new NonconservativeDGTraceIntegrator(velocity, -1.0, -0.5));
```

Comment:
Maybe define `alpha` and `beta` if we want to make things even more clean and clear:
```c++
const double alpha = -1.0;
const double beta = -0.5;
k->AddDomainIntegrator(new ConvectionIntegrator(velocity, alpha));
k->AddInteriorFaceIntegrator(
   new NonconservativeDGTraceIntegrator(velocity, alpha, beta));
k->AddBdrFaceIntegrator(
   new NonconservativeDGTraceIntegrator(velocity, alpha, beta));
```

Reply:
It would also be tempting to do:
```c++
k->AddDomainIntegrator(new NonconservativeDGConvectionIntegrator(velocity));
k->AddInteriorFaceIntegrator(
   new NonconservativeDGTraceIntegrator(velocity));
k->AddBdrFaceIntegrator(
   new NonconservativeDGTraceIntegrator(velocity));
```
and have
```c++
class NonconservativeDGConvection: public ConvectionIntegrator
{
public:
   NonconservativeDGConvection(VectorCoefficient &velocity): ConvectionIntegrator(velocity, -1.0) { }
};
```

Reply:
May want to use `constexpr` for constants but I like the idea of spelling out what's going on.

> ```c++
> constexpr double alpha = -1.0;
> constexpr double beta = -0.5;
> k->AddDomainIntegrator(new ConvectionIntegrator(velocity, alpha));
> k->AddInteriorFaceIntegrator(
>    new NonconservativeDGTraceIntegrator(velocity, alpha, beta));
> k->AddBdrFaceIntegrator(
>    new NonconservativeDGTraceIntegrator(velocity, alpha, beta));
> ```

Reply:
I like the idea of default values for the common use case but I think the default values could be better implemented.

> ```c++
> class NonconservativeDGConvection: public ConvectionIntegrator
> {
> public:
>    NonconservativeDGConvection(VectorCoefficient &velocity, double alpha=-1.0) /* ... */
> };
> ```

Reply:
I added constexpr values for alpha and beta to ex9 and ex9p. See [the other comment](https://github.com/mfem/mfem/pull/1965#discussion_r551526492) for discussion on default values.